constants/src/lib.rs

```diff
@@ -31,3 +46,136 @@ pub const INVALID_CHARACTER_VALUE: &str = "22018";
 pub const SQL_ALL_TABLE_TYPES: &str = "%";
 pub const SQL_ALL_CATALOGS: &str = "%";
 pub const SQL_ALL_SCHEMAS: &str = "%";
+
+fn format_driver_version() -> String {
+    // The driver version can be obtained from the Cargo.toml file.
+    // The env! macro call below gets the version from the Cargo file
+    // at compile time.
+    let version_major = env!("CARGO_PKG_VERSION_MAJOR");
+    let version_minor = env!("CARGO_PKG_VERSION_MINOR");
+    let version_patch = env!("CARGO_PKG_VERSION_PATCH");
+
+    format_version(version_major, version_minor, version_patch)
+}
+
+fn format_version(major: &str, minor: &str, patch: &str) -> String {
+    format!(
+        "{}.{}.{}",
+        format_version_part(major, 2),
+        format_version_part(minor, 2),
+        format_version_part(patch, 4)
+    )
+}
+
+fn format_version_part(part: &str, len: usize) -> String {
+    if len < part.len() {
+        return part.to_string();
+    }
+    format!("{}{}", "0".repeat(len - part.len()), part)
+}
+
+mod unit {
+    #[cfg(test)]
+    use super::format_version;
+
+    macro_rules! format_version_test {
+        ($func_name:ident, expected = $expected:expr, major = $major:expr, minor = $minor:expr, patch = $patch:expr) => {
+            #[test]
+            fn $func_name() {
+                let actual = format_version($major, $minor, $patch);
+                assert_eq!($expected, actual)
+            }
+        };
+    }
+
+    format_version_test!(
+        no_padding_needed,
+        expected = "10.11.1213",
+        major = "10",
+        minor = "11",
+        patch = "1213"
+    );
+
+    format_version_test!(
+        padding_needed,
+        expected = "01.01.0001",
+        major = "1",
+        minor = "1",
+        patch = "1"
+    );
+
+    format_version_test!(
+        parts_larger_than_length,
+        expected = "111.222.33333",
+        major = "111",
+        minor = "222",
+        patch = "33333"
+    );
+
+    format_version_test!(
+        format_cargo_version,
+        expected = "00.01.0000",
+        major = "0",
+        minor = "1",
+        patch = "0"
+    );
+}
+
+#[cfg(test)]
+mod version {
```

Comment:
@nbagnard I thought more about the differing versions potential problem. This test ensures that all versions remain in sync.